src/AutoUI/index.tsx

```diff
@@ -260,36 +307,39 @@ export const AutoUI = <T extends AutoUIBaseResource<T>>({
 
 	React.useEffect(() => {
 		$setSelected([], 'none');
-	}, [filters]);
-
-	const onActionTriggered = React.useCallback((actionData: ActionData<T>) => {
-		setActionData(actionData);
-		if (actionData.action.actionFn) {
-			actionData.action.actionFn({
-				affectedEntries: actionData.affectedEntries,
-				checkedState: checkedState || 'none',
-				setSelected: $setSelected,
-			});
-		}
-	}, []);
+	}, [filters, $setSelected]);
+
+	const onActionTriggered = React.useCallback(
+		(acData: ActionData<T>) => {
+			setActionData(acData);
+			if (acData.action.actionFn) {
+				acData.action.actionFn({
+					affectedEntries: acData.affectedEntries,
+					checkedState: checkedState || 'none',
+					setSelected: $setSelected,
+				});
+			}
+		},
+		[$setSelected, checkedState],
+	);
 
 	const defaultLensSlug = getFromLocalStorage(`${model.resource}__view_lens`);
 
 	const lenses = React.useMemo(
-		() => getLenses(data, lensContext, customLenses),
+		() => getLenses<T>(data, lensContext, customLenses),
 		[data, lensContext, customLenses],
 	);
 
 	const [lens, setLens] = React.useState<LensTemplate>(lenses[0]);
 
 	React.useEffect(() => {
 		const foundLens =
-			lenses.find((lens) => lens?.slug === defaultLensSlug) || lenses[0];
+			lenses.find((l) => l?.slug === defaultLensSlug) ?? lenses[0];
```

Comment:
Do you remember why we do it like this in a useEffect instead of putting this directly in the useState as the default value?